android/src/main/java/com/novoda/notils/devicedetection/AndroidVersion.java

```diff
@@ -0,0 +1,50 @@
+package com.novoda.notils.devicedetection;
+
+import android.os.Build;
+
+public class AndroidVersion {
+
+    private final int osApiLevel;
+    private final String osVersionName;
+
+    public static AndroidVersion newInstance() {
+        return new AndroidVersion(Build.VERSION.SDK_INT, Build.VERSION.RELEASE);
+    }
+
+    AndroidVersion(int osApiLevel, String osVersionName) {
+        this.osApiLevel = osApiLevel;
+        this.osVersionName = osVersionName;
+    }
+
+    public boolean is18JellyBeanOrOver() {
+        return osApiLevel >= Build.VERSION_CODES.JELLY_BEAN_MR2;
+    }
+
+    public boolean is19KitKat() {
+        return osApiLevel == Build.VERSION_CODES.KITKAT;
+    }
+
+    public boolean is19KitKatOrOver() {
+        return osApiLevel >= Build.VERSION_CODES.KITKAT;
+    }
+
+    public boolean is21LollipopOrOver() {
+        return osApiLevel >= Build.VERSION_CODES.LOLLIPOP;
+    }
+
+    public boolean isPre21Lollipop() {
+        return !is21LollipopOrOver();
+    }
+
+    public boolean is23MarshmallowOrOver() {
+        return osApiLevel >= Build.VERSION_CODES.M;
```

Comment:
When you have `if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M)` in code, android lint detects it and does not warn api level related stuff that requires `M` inside that if statement. 

But if you wrap it like this, unfortunately that link check doesn't detect and you will end-up using `@TargetApi` annotation to suppress the warning. 

`@TargetApi` means that the method/class is targeted only to that api level. But in most cases they are not. They are used for all devices but with slight modification on some api levels.